data/src/main/java/io/gnosis/data/repositories/TransactionRepository.kt

```diff
@@ -76,35 +76,37 @@ class TransactionRepository(
                 SafeRepository.isSettingsMethod(transactionDto.dataDecoded?.method)
 
     // This is a big assumption for txType == ETHEREUM_TRANSACTION, it was agreed that this can be assumed successful, because only successful TXs trigger events
-    private fun transfer(transferDto: TransferDto): Transaction.Transfer {
-        val tokenInfo = serviceTokenInfo(transferDto)
+    private fun transfer(transfer: TransferDto, safeInfo: SafeInfo): Transaction.Transfer {
```

Comment:
Here you are forced to to pass all the way down to the lowest level of mapping the `SafeInfo` which, if you did in the UI layer, you could simply call the method. I think it would be a cleaner solution to have it as an extension function like I mentioned on my previous comment.